### PR TITLE
Update ReplaceBranch hook to avoid running on `--amend` by default

### PR DESCRIPTION
**Pull request by jethrodaniel**
- changes default to not run on `--amend`
- changes the default pattern
- improves the documentation for this hook
- refactors this hook's specs to reduce mocking and increase coverage

### CODE REVIEW

**Comment by jethrodaniel** [comment]
Annoyingly, CI is complaining about a rubocop violation that doesn't exist. likely due to my force-pushes on this branch.

File: lib/overcommit/hook/prepare_commit_msg/replace_branch.rb [hunk 1]
```diff
@@ -17,21 +43,21 @@ def run
...
-        "#{new_template} #{old_contents}"
+        "#{new_template}#{old_contents}"
       end
 
       :pass
     end
 
     def new_template
-      @new_template ||= Overcommit::GitRepo.current_branch.gsub(branch_pattern, replacement_text)
+      @new_template ||= Overcommit::GitRepo.current_branch.gsub(branch_pattern, replacement_text).strip
```
Comment (sds) [inline lib/overcommit/hook/prepare_commit_msg/replace_branch.rb:53]: I believe RuboCop is complaining about this line, @jethrodaniel. Should be easy to address.

**Comment by sds** [comment]
Overall looks great. Once we get that one RuboCop line fixed we'll be good to merge. Thank you!

**Comment by jethrodaniel** [comment]
Thanks - seems I was using rubocop with my global config, instead of this project's config.

**Comment by sds** [comment]
Thank you!

**Pull request merged**